features/step_definitions/redmine_steps.rb

```diff
@@ -73,12 +78,22 @@
 
 When(/^I try to open random project with (\d+) retries$/) do |retries_count|
   @project[:name] = Faker::Hipster.word.capitalize
-  open_random_project(@project[:name], retries_count.to_i)
+  random_project_url = "http://demo.redmine.org/projects/#{@project[:name]}"
+
+  i = 0
```

Comment:
Now separate method `open_url_with_retries` added for this but some logic is still here, in step definition so step definition looks like:

```ruby
When(/^I try to open random project with (\d+) retries$/) do |retries_count|
  random_name = Faker::Hipster.word.capitalize
  random_project_url = BasicPage::BASIC_URL + '/projects/' + random_name

  project_exists = open_url_with_retries(random_project_url, retries_count.to_i)

  if project_exists
    puts "Project with name [#{random_name}] already exists"
  else
    @project.name = random_name
    visit(CreateProjectPage).create_project(@project)
    puts "New project [#{@project.name}] created"
  end
end
```

I've thought about making it shorter, but see no obvious way how to hande this without affecting readability, what do you think about this?